ui/raidboss/emulator/overrides/RaidEmulatorTimelineUI.js

```diff
@@ -0,0 +1,143 @@
+    this.$progressTemplate = document.querySelector('template.progress').content.firstElementChild;
+  }
+
+  bindTo(emulator) {
+    emulator.on('tick', (timestampOffset, lastLogTimestamp) => {
+      this.emulatedTimeOffset = timestampOffset;
+      for (let i in this.emulatedTimerBars) {
+        let bar = this.emulatedTimerBars[i];
+        this.updateBar(bar, timestampOffset);
+      }
+      let toRemove = this.emulatedTimerBars.filter((bar) => bar.forceRemoveAt <= timestampOffset);
+      for (let i in toRemove) {
+        let bar = toRemove[i];
+        bar.$progress.remove();
+      }
+      this.emulatedTimerBars = this.emulatedTimerBars.filter((bar) => {
+        return bar.forceRemoveAt > timestampOffset;
+      });
+      this.timeline && this.timeline.timebase && this.timeline._OnUpdateTimer();
+    });
+    emulator.on('play', () => {
+      this.emulatedStatus = 'play';
+      this.timeline && this.timeline.emulatedSync(emulator.currentTimestamp);
+    });
+    emulator.on('pause', () => {
+      this.emulatedStatus = 'pause';
+    });
+    let tmpPopupText;
+    emulator.on('preSeek', (time) => {
+      this.timeline && this.timeline.Stop();
+      this.emulatedTimeOffset = time;
+      for (let i in this.emulatedTimerBars) {
+        let bar = this.emulatedTimerBars[i];
+        bar.$progress.remove();
+      }
+      this.emulatedTimerBars = [];
+      this.timeline && (tmpPopupText = this.timeline.popupText);
+      this.timeline && (this.timeline.popupText = null);
+    });
+    emulator.on('postSeek', (time) => {
+      this.timeline && (this.timeline.popupText = tmpPopupText);
+      this.timeline && this.timeline.emulatedSync(time);
+      for (let i in this.emulatedTimerBars) {
+        let bar = this.emulatedTimerBars[i];
+        this.updateBar(bar, time);
+      }
+    });
+    emulator.on('currentEncounterChanged', () => {
+      this.timeline && this.timeline.Stop();
+      this.emulatedTimeOffset = 0;
+      for (let i in this.emulatedTimerBars) {
+        let bar = this.emulatedTimerBars[i];
+        bar.$progress.remove();
+      }
+      this.emulatedTimerBars = [];
+    });
+  }
+
+  updateBar(bar, timestampOffset) {
+    let barElapsed = timestampOffset - bar.start;
+    let barProg = Math.min((barElapsed / bar.duration) * 100, 100);
+    if (bar.style === 'empty')
+      barProg = 100 - barProg;
+
+    let rightText = ((bar.duration - barElapsed) / 1000).toFixed(1);
+    if (barProg >= 100)
+      rightText = '';
+
+    bar.$leftLabel.textContent = bar.event.text;
+    bar.$rightLabel.textContent = rightText;
+    bar.$bar.ariaValueNow = barElapsed;
+    bar.$bar.style.width = barProg + '%';
+  }
+
+  // Overrride
+  Init() {
+    if (this.init)
+      return;
+    this.init = true;
+    this.timerlist = { clear: () => { } };
+
+    this.activeBars = {};
+    this.expireTimers = {};
+  }
+
+  // Overrride
+  InitDebugUI() {
+    this.debugElement = {};
```

Comment:
Ahhh, I misunderstood what this bug was.  `debugElement` expects to be an element.  I think this needs to be `this.debugElement = document.createElement('div');` otherwise you get a bunch of errors when `Options.Debug` is true.

Reply:
I've made that adjustment but realistically we should probably display that div in some manner in the future I think.